File: Ticket-order-Mikhail/TicketOrder/src/main/java/com/services/FilmService.java
package com.services;

import com.entities.Film;
import com.entities.Hall;
import com.entities.Rating;
import com.repositories.FilmRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.time.Instant;
import java.util.Date;

@Service
@ComponentScan("com")
public class FilmService {

    private final FilmRepository filmRepository;

    @Autowired
    public FilmService(FilmRepository filmRepository) {
        this.filmRepository = filmRepository;
    }

    @Transactional
    public Film someWork() {
        Rating rating = new Rating();
        rating.upgrade(8d);
        Film film = filmRepository.save(new Film("title", "description", rating , Date.from(Instant.parse("2018-10-02"))));
        return film;
    }
}

File: main/java/com/services/FilmService.kt
package com.services

import com.entities.Film
import com.entities.Rating
import com.repositories.FilmRepository
import org.springframework.data.domain.PageRequest
import org.springframework.stereotype.Service
import javax.persistence.EntityNotFoundException

@Service
class FilmService(
        private val filmRepository: FilmRepository,
        private val visitorService: VisitorService
) {

    fun findByTitlePart(titlePart: String, page: Int, size: Int): List<Film> {
        return filmRepository.findAllByTitleStartingWith(PageRequest.of(page - 1, size), titlePart)
    }

    fun list(page: Int, size: Int): List<Film> {
        return filmRepository.findAllOrderByRatingDesc(PageRequest.of(page - 1, size))
    }

    @Throws(EntityNotFoundException::class)
    fun upgradeRating(filmId: Long, rating: Rating): Film {
        val film = filmRepository.findById(filmId).orElseThrow {
            EntityNotFoundException(
                    "Film with id : $filmId does`t found"
            )
        }
        film.getRating().upgrade(rating.getValue())
        return filmRepository.save(film)
    }
}
